### PR TITLE
Fixes #611 LSP4Jakarta ambiguously matches the expected class names when checking diagnostics, applying quick fixes, etc

### LINKED ISSUE #611: Found LSP4Jakarta ambiguously matches the expected class names when checking diagnostics, applying quick fixes, etc ...
Found LSP4Jakarta unambiguous matches, the expected class names when checking diagnostics, applying quick fixes, etc... (e.g. @jakarta.ws.rs.PUT vs @my.random.pkg.PUT).

Screenshot 2024-01-01 at 2 05 03 PM (1)

In this image, we observe repeated annotations, with one is from "Jakarta." and another from "Javax.". As a temporary solution, we filtered out names starting with "Jakarta."

Screenshot 2024-01-09 at 6 26 59 PM

In this image, there are still unambiguous matches even though we filtered out names that begin with "Jakarta."

Mentioning this issue that opened in lsp4jakarta --> https://github.com/eclipse/lsp4jakarta/issues/507

### PR DESCRIPTION
**Pull request by aparnamichael**
Fixes #611

LSP4Jakarta ambiguously matches the expected class names when checking diagnostics, applying quick fixes, etc

Updated all of the diagnostic collectors to include fully qualified class names in the diagnostic data that they produce. And fixed the quick fix  issue of JsonbTransientAnnotationQuickFix and ScopeDeclarationQuickFix.

### CODE REVIEW

File: src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/jsonb/JsonbDiagnosticsCollector.java
Comment (vaisakhkannan) [inline src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/jsonb/JsonbDiagnosticsCollector.java:1]: * Copyright (c) 2020, 2023 IBM Corporation, Matheus Cruz and others. 

Update the year to 2024

**Review by vaisakhkannan — CHANGES_REQUESTED** [review]

File: src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/jsonb/JsonbDiagnosticsCollector.java
Comment (aparnamichael) [reply src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/jsonb/JsonbDiagnosticsCollector.java:1]: done

**Review by vaisakhkannan — DISMISSED** [review]
looks good to me!

File: src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/cdi/ManagedBeanDiagnosticsCollector.java [hunk 1]
```diff
@@ -108,12 +105,10 @@ else if (INJECT_FQ_NAME.equals(annotation))
                         isInjectField = true;
                 }
                 if (isProducerField && fieldScopes.size() > 1) {
-                    List<String> diagnosticData = fieldScopes.stream().map(annotation -> getSimpleName(annotation))
-                            .collect(Collectors.toList()); // convert to simple name
-                    diagnosticData.add(PRODUCES);
+                    fieldScopes.add(PRODUCES);
```
Comment (mrglavas) [inline src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/cdi/ManagedBeanDiagnosticsCollector.java:108]: Ensure that all names in the list are qualified names.

File: src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/cdi/ManagedBeanDiagnosticsCollector.java [hunk 2]
```diff
@@ -167,12 +162,10 @@ else if (INJECT_FQ_NAME.equals(annotation))
                 }
 
                 if (isProducerMethod && methodScopes.size() > 1) {
-                    List<String> diagnosticData = methodScopes.stream().map(annotation -> getSimpleName(annotation))
-                            .collect(Collectors.toList()); // convert to simple name
-                    diagnosticData.add(PRODUCES);
+                    methodScopes.add(PRODUCES);
```
Comment (mrglavas) [inline src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/cdi/ManagedBeanDiagnosticsCollector.java:165]: Ensure that all names in the list are qualified names.

File: src/test/java/io/openliberty/tools/intellij/lsp4jakarta/it/cdi/ManagedBeanTest.java [hunk 3]
```diff
@@ -104,46 +104,92 @@ public void scopeDeclaration() throws Exception {
         Diagnostic d1 = d(12, 16, 17,
                 "Scope type annotations must be specified by a producer field at most once.",
                 DiagnosticSeverity.Error, "jakarta-cdi", "InvalidScopeDecl");
-        d1.setData(new Gson().toJsonTree(Arrays.asList("Dependent", "ApplicationScoped", "Produces")));
+        d1.setData(new Gson().toJsonTree(Arrays.asList("jakarta.enterprise.context.Dependent",
+                "jakarta.enterprise.context.ApplicationScoped", "Produces")));
```
Comment (mrglavas) [inline src/test/java/io/openliberty/tools/intellij/lsp4jakarta/it/cdi/ManagedBeanTest.java:108]: This expected output is incorrect. All class names in the list should be fully qualified.

File: src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/jsonb/JsonbTransientAnnotationQuickFix.java [hunk 4]
```diff
@@ -35,20 +36,15 @@ public class JsonbTransientAnnotationQuickFix extends RemoveMultipleAnnotations
...
-        annotations.remove(JsonbConstants.JSONB_TRANSIENT);
+        annotations.remove(JsonbConstants.JSONB_TRANSIENT_FQ_NAME);
         if (annotations.size() > 0) {
-            // Convert the short annotation names to their fully qualified equivalents.
-            List<String> fqAnnotations = new ArrayList<>();
-            for (String annotation : annotations) {
-                fqAnnotations.addAll(getFQAnnotationNames(project, annotation));
-            }
-            annotationsListsToRemove.add(fqAnnotations);
+            annotationsListsToRemove.addAll(Collections.singleton(annotations));
```
Comment (mrglavas) [inline src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/jsonb/JsonbTransientAnnotationQuickFix.java:47]: This seems to do the same thing as the old line of code but is less efficient and harder to understand. Why wrap the list in a collection?

File: src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/cdi/ManagedBeanDiagnosticsCollector.java (hunk 2 above)
Comment (aparnamichael) [reply src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/cdi/ManagedBeanDiagnosticsCollector.java]: Done

File: src/test/java/io/openliberty/tools/intellij/lsp4jakarta/it/cdi/ManagedBeanTest.java (hunk 3 above)
Comment (aparnamichael) [reply src/test/java/io/openliberty/tools/intellij/lsp4jakarta/it/cdi/ManagedBeanTest.java]: Done

File: src/test/java/io/openliberty/tools/intellij/lsp4jakarta/it/cdi/ManagedBeanTest.java [hunk 5]
```diff
@@ -104,46 +104,92 @@ public void scopeDeclaration() throws Exception {
...
-        d2.setData(new Gson().toJsonTree(Arrays.asList("ApplicationScoped", "RequestScoped", "Produces")));
-        
-        Diagnostic d3 = d(10, 13, 29, "Scope type annotations must be specified by a managed bean class at most once.",
-                DiagnosticSeverity.Error, "jakarta-cdi", "InvalidScopeDecl");
-        d3.setData(new Gson().toJsonTree(Arrays.asList("ApplicationScoped", "RequestScoped")));
+        Diagnostic d2 = d(15, 25, 41, "Scope type annotations must be specified " +
+                        "by a producer method at most once.", DiagnosticSeverity.Error, "jakarta-cdi",
+                "InvalidScopeDecl");
+        d2.setData(new Gson().toJsonTree(Arrays.asList("jakarta.enterprise.context.ApplicationScoped",
+                "jakarta.enterprise.context.RequestScoped", "Produces")));
```
[the remark this answers is not in the document]
Comment (aparnamichael) [reply src/test/java/io/openliberty/tools/intellij/lsp4jakarta/it/cdi/ManagedBeanTest.java]: Done

File: src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/jsonb/JsonbTransientAnnotationQuickFix.java (hunk 4 above)
Comment (aparnamichael) [reply src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/jsonb/JsonbTransientAnnotationQuickFix.java]: Done

**Review by vaisakhkannan — APPROVED** [review]
Looks good

**Comment by aparnamichael** [comment]
@mrglavas I have addressed all the review comments.

Also I have gone through all of the diagnostic classes and all of them are child class of AbstractDiagnosticsCollector. And all of these classes are using fully qualified name.

**Pull request merged**